src/main.rs

```diff
@@ -48,124 +63,196 @@ impl LogLevel {
 
 #[derive(Subcommand)]
 enum Commands {
-    Check,
-    Stamp,
+    HealthCheck,
+    RollBack,
 }
 
-#[derive(Serialize, Deserialize, PartialEq, Eq, Hash)]
-struct ServiceStatus {
-    unit: String,
-}
+fn run_diagnostics() -> Result<(), Error> {
+    let mut script_failure: bool = false;
+    let mut path_exists: bool = false;
+    for path in GREENBOOT_INSTALL_PATHS {
+        let gereenboot_required_path = format!("{}{}", path, "/check/required.d");
+        if Path::new(&gereenboot_required_path).is_dir() {
+            path_exists = true;
 
-fn check() -> Result<(), Error> {
-    match File::open("/etc/greenboot/upgrade.stamp") {
```

Comment:
if you remove this it doesn't get only on upgrade as we discussed but why would that be? if you look at this stamp file + the Condition in the service file this is about avoid that cleanly

Reply:
I am actually tackling the health-check first. which is upto set in the boot_counter and boot_success. so removed most of the things to de-clutter. I will add this bits when handling the rollback.

Reply:
Actually the approach changed with the current implementation as it will run health-check on every reboot instead of only checking when its upgraded.